public/API/API_GetComments.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+use App\Models\Achievement;
+use App\Models\Comment;
+use App\Models\User;
+use App\Policies\CommentPolicy;
+use App\Support\Rules\CtypeAlnum;
+use Illuminate\Support\Facades\Validator;
+
+/*
+*  API_GetComments - returns the comments associated to a game or achievement
+*    i : game or achievement id
+*    u : username
+*    t : 1 = game, 2 = achievement, 3 = user
```

Comment:
Doing a final review now. This took a minute to put my thoughts together on - apologies for it coming so late in the review. I promise this is the last thing.

The final piece that's nagging at me is potential for improvement w/ the DX of these query params.

I've been thinking through how I'm going to implement this endpoint in the JS client lib, and the ergonomics of the `t` query param keeps leaping off the page at me.

I'm considering the shape of one of these calls:
```
GET /API/API_GetComments.php?u=iOSLife&t=3
```
This seems quite duplicative to me. Because `iOSLife` is a string and it can only be provided via `u`, I can't help but feel the endpoint should not need to be reminded of this by means of `t=3`.

Logically, an improvement would be to make `t` optional when `u` is given. However, this also begs the question of whether `u` and `i` can thus be coalesced into a single query parameter. I think the best possible DX looks something like:

```bash
GET /API/API_GetComments.php?i=iOSLife # i! this just works
GET /API/API_GetComments.php?i=iOSLife&t=3 # this also works
GET /API/API_GetComments.php?i=15502&t=1 # this works too
# no more u param. consumers can just work with i and t.
```

Even though it's not an ID, we absorb the username into `i`. `t` can still be provided (it's safe to assume some consumers will have libraries that do this eagerly), but is optional & implied if `i` is a string. `t` is only required if an integer is given for `i`, as when this occurs, the endpoint cannot reasonably determine what resource the user is trying to fetch.

Reply:
I like this a lot and will work on the change. Thanks for working it out and as always, never a rush on the reviews.

Reply:
Just pushed up changes to address this. Let me know your thoughts.

I also caught that banned users could call this endpoint and get _some_ results back, so I added a check to ensure the calling user is not banned. I'm not sure if we invalidate API keys once a user is banned, if we do I'll revert that.